src/checkbox/checkbox.js

```diff
@@ -15,255 +15,214 @@
  * limitations under the License.
  */
 
-(function() {
-  'use strict';
-
+/**
+ * The MaterialCheckbox class wraps a Material Design checkbox component.
+ *
+ * @export
+ */
+class MaterialCheckbox {
   /**
-   * Class constructor for Checkbox MDL component.
-   * Implements MDL component design pattern defined at:
-   * https://github.com/jasonmayes/mdl-component-design-pattern
+   * Initialize checkbox from a DOM node.
    *
-   * @constructor
-   * @param {HTMLElement} element The element that will be upgraded.
-   */
-  var MaterialCheckbox = function MaterialCheckbox(element) {
-    this.element_ = element;
+   * @param {Element} root The element being upgraded.
+   */
+  constructor(root) {
+    // Check if the root has the right class.
+    this.root_ = root;
+    if (!root.classList.contains(MaterialCheckbox.classes_.ROOT)) {
```

Comment:
Yes, I think there's a lot we'll be able to do with a base class, including things like the `initComponents` work too. Once everyone approves the POC, I'll expand it to include several components, and I'll generalise as much behaviour as possible into a base class.